backend/models/task_detail.py

```
@@ -1,10 +1,59 @@
 from db.migrations.create_table import db
+from models.user import UserModel
+from flask_jwt_extended import jwt_required, get_jwt_identity
 
-class ItemModel(db.Model):
+class TaskDetailModel(db.Model):
     __tablename__ = "task_details"
 
     id = db.Column(db.Integer, primary_key=True)
     start_time = db.Column(db.BigInteger)
     finish_time = db.Column(db.BigInteger)
-    user_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+    task_name = db.Column(db.String)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
     user = db.relationship("UserModel")
+
+    def __init__(self, user_id, task_name, start_time, finish_time = None):
+        self.user_id = user_id
+        self.task_name = task_name
+        self.start_time = start_time
+        self.finish_time = finish_time
+    
+    @classmethod
+    def find_all(cls):
+        return cls.query.all()
+    
+    @classmethod
+    def find_by_name(cls, taskName):
+        return cls.query.filter_by(task_name=taskName).first()
+
+    @classmethod
+    def is_task_finished(cls) -> bool:
+        records = cls.query.all()
+        for record in records:
+            if record.finish_time is None:
+                return False
+        return True
+    
+    def save_to_db(self):
+        db.session.add(self)
+        db.session.commit()
+
+    def json(self):
+        return {
+            "id" : self.id,
+            "user_id" : self.user_id,
+            "task_name" : self.task_name,
+            "start_time" : self.start_time,
+            "finish_time" : self.finish_time
+        }
+
+    @classmethod
+    @jwt_required()
+    def find_current_user(cls):
```

this block should throw an error that should be caught by try and catch block by the caller